### PR TITLE
Ms1 fixes

### PR DESCRIPTION
**Pull request by almogdepaz**

### CODE REVIEW

File: hare/hare.go [hunk 1]
```diff
@@ -160,9 +160,9 @@ func (h *Hare) onTick(id mesh.LayerID) {
 	}
 
 	// retrieve set form orphan blocks
-	blocks := h.obp.GetUnverifiedLayerBlocks(h.lastLayer)
-	if len(blocks) == 0 {
-		log.Info("No blocks for consensus on layer %v", id)
+	blocks, err := h.obp.GetUnverifiedLayerBlocks(h.lastLayer)
+	if err != nil {
+		log.Info("No blocks for consensus on layer %v %v", id, err)
```
Comment (zalmen) [inline hare/hare.go]: Why Info? Can it occur in a non-faulty scenario?

File: mesh/mesh.go [hunk 2]
```diff
@@ -47,8 +48,11 @@ func NewMesh(layers, blocks, validity database.DB, mesh MeshValidator, state Sta
 		Log:      logger,
 		tortoise: mesh,
 		state:    state,
-		meshDB:   NewMeshDB(layers, blocks, validity),
+		done:     make(chan struct{}),
+		meshDB:   NewMeshDB(layers, blocks, validity, logger),
 	}
+
+	//go ll.onTick()
```
Comment (zalmen) [inline mesh/mesh.go]: remove?

File: sync/syncer.go [hunk 3]
```diff
@@ -76,41 +78,40 @@ func (s *Syncer) Start() {
...
-					s.Synchronise()
-					atomic.StoreUint32(&s.SyncLock, IDLE)
-				}
-			}()
+		case <-s.forceSync:
+			go foo()
+		case layer := <-s.clock:
+			atomic.StoreUint32(&s.currentLayer, uint32(layer))
+			s.Info("sync got tick for layer %v", layer)
+			go foo()
```
Comment (zalmen) [inline sync/syncer.go:97]: Just to make sure I understand - this is where layers are pushed to the Tortoise, right? 
If so, why do you wait till the beginning of the layer in order to push the previous layer to the tortoise? You can do that after a minute (or less) from the beginning of the previous layer when all the honest blocks had arrived. 
Another "just to make sure" - this doesn't handle late blocks, right? In the current state of the code late blocks will be ignored?

Comment (almogdepaz) [reply sync/syncer.go:97]: you got it right, late block handling will be add with #565 
and the concurrent handling will also be a separate pr

File: hare/hare.go (hunk 1 above)
Comment (almogdepaz) [reply hare/hare.go]: legacy code...

**Review by zalmen — APPROVED** [review]

**Pull request merged**